src/main/java/com/google/flourbot/api/CardResponse.java

```diff
@@ -0,0 +1,96 @@
+import com.google.api.services.chat.v1.model.OnClick;
+import com.google.api.services.chat.v1.model.OpenLink;
+import com.google.api.services.chat.v1.model.Section;
+import com.google.api.services.chat.v1.model.TextButton;
+import com.google.api.services.chat.v1.model.TextParagraph;
+import com.google.api.services.chat.v1.model.WidgetMarkup;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+public class CardResponse {
+
+  private static final String INTERACTIVE_TEXT_BUTTON_ACTION = "doTextButtonAction";
+  private static final String INTERACTIVE_IMAGE_BUTTON_ACTION = "doImageButtonAction";
+  private static final String INTERACTIVE_BUTTON_KEY = "originalMessage";
+  private static final String HEADER_IMAGE = "https://shop.googlemerchandisestore.com/store/20160512512/assets/items/largeimages/GGOEACBA116699.jpg";
+  private static final String BOT_NAME = "MacroBot";
+  private static final String REDIRECT_URL = "https://goo.gl/kwhSNz"; // TODO: Change to spreadsheet link
+
+  /**
+    * Creates a card-formatted response based on the message sent in Hangouts Chat.
+    *
+    * @param message the event object sent from Hangouts Chat
+    * @return a card instance
+    */
+  public static Card createCardResponse(String message) {
+    Card card = new Card();
+    List<WidgetMarkup> widgets = new ArrayList<>();
+    List<ActionParameter> customParameters = Collections.singletonList(
+            new ActionParameter().setKey(INTERACTIVE_BUTTON_KEY).setValue(message)
+    );
+
+    // HEADER
+    CardHeader header = new CardHeader()
+                  .setTitle(BOT_NAME)
+                  .setSubtitle("Card header")
+                  .setImageUrl(HEADER_IMAGE)
+                  .setImageStyle("IMAGE");
+    card.setHeader(header);
+
+    // TEXT PARAGRAPH
+    TextParagraph textParagraphWidget = new TextParagraph().setText("<b>The reply text we got:</b> is <br></br>" + message);
+    widgets.add(new WidgetMarkup().setTextParagraph(textParagraphWidget));
+
+    // TODO: Add cute image if CongraBot (do this after we make a response object)
+
+    // KEY VALUE
+    KeyValue keyValueWidget = new KeyValue()
+                  .setTopLabel("KeyValue widget")
+                  .setContent("This is a KeyValue widget")
+                  .setBottomLabel("The bottom label")
+                  .setIcon("DESCRIPTION");
+    widgets.add(new WidgetMarkup().setKeyValue(keyValueWidget));
+
+    // TEXT BUTTON
+    OpenLink openLink = new OpenLink().setUrl(REDIRECT_URL);
```

Comment:
There is a lot of things added here, it seems that we need some conditions to apply only relevant fields, correct?